### PR TITLE
Update grid layout

### PR DESCRIPTION
**Pull request by HanLee25**
This commit replaces `float` properties with `grid` layout to support
better responsive design and reduce redundant `div` wrappers.

Future work:
* reduce the use of `clearfix` if possible

Before:
image

After:
image

### CODE REVIEW

File: stylesheets/modules/_footer.scss [hunk 1]
```diff
@@ -9,9 +9,15 @@
...
     @include clearfix();
+    display: grid;
+    grid-template-columns: 1fr;
     font-size: 1.25em;
     margin-bottom: $gutter-width;
-    max-width: 22em;
+
+    @media screen and (min-width : $medium-screen) {
+      grid-template-columns: auto 1fr;
+      max-width: 22em;
```
Comment (keiranking) [inline stylesheets/modules/_footer.scss]: I notice use of 22em in the `_header` and `_footer` modules. Consider creating a Sass variable to make this code less brittle.

File: stylesheets/components/_dangerdanger.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.dangerdanger {
```
Comment (keiranking) [inline stylesheets/components/_dangerdanger.scss]: Not sure of the relationship between classes `dangerdanger`, `references-dangerdanger`, `site-header-dangerdanger`, `references`, and `site-header`. If there is a bunch of refactoring work to be done, that could be its own PR.

File: stylesheets/modules/_footer.scss (hunk 1 above)
Comment (HanLee25) [reply stylesheets/modules/_footer.scss]: Good point. And it gave me another chance to look at these lines.

Now that I look at them again, it seems `max-width` can be replaced by a size range of `grid-template-columns` since the elements are grid items. And that ends up with different values of size for each grid layout which demotivates me to create variables for.

File: stylesheets/components/_dangerdanger.scss (hunk 2 above)
Comment (HanLee25) [reply stylesheets/components/_dangerdanger.scss]: This change allows me to separate layout properties that belong to the wrapper element and the visual part that belong to SVG itself. I guess better convention would be helpful, but this separation would make more value with this PR I feel.
